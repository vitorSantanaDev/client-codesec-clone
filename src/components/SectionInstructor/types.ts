import { ISectionInstructorProps } from 'interfaces/sections'

export type IViewProps = ISectionInstructorProps
