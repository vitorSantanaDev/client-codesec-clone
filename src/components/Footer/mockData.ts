export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
