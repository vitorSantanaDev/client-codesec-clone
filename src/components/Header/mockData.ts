import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
