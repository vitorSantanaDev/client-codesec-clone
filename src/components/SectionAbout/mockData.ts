export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
