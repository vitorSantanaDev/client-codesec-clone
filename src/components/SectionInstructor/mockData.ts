export const mockDataSectionInstructor = {
  title: 'Title Example Instructor Section',
  subtitle: 'Subtitle',
  instructor: {
    name: 'Name of Instructor',
    photoUrl: '',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.',
    link: {
      linkText: 'Saiba mais',
      linkUrl: 'https://google.com.br'
    }
  }
}
