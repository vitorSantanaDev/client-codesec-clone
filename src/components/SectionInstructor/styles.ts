import styled, { css } from 'styled-components'
import { Wrapper as Heading } from 'components/Heading/styles'
import { Wrapper as Subtitle } from 'components/Subtitle/styles'

type PhotoContainerProps = {
  urlImage: string
}

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} 0;

    > ${Subtitle} {
      margin-bottom: calc(${theme.spacings.small} - 0.8rem);
    }

    @media ${theme.media.lteMedium} {
      > ${Heading} {
        font-size: calc(${theme.font.sizes.large} + 0.4rem);
      }
    }
  `}
`
export const InstructorInfoWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;

    margin-top: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      justify-content: center;
    }
  `}
`
export const InstructorData = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: ${theme.spacings.medium};
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;
    }
  `}
`
export const InstructorPhoto = styled.div<PhotoContainerProps>`
  ${({ theme, urlImage }) => css`
    width: 120px;
    height: 120px;
    border-radius: 50%;
    border: 2px solid
      ${urlImage === '/images//images/icon-autor.svg'
        ? `${theme.colors.redPrimary}`
        : null};

    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
    background-image: url('${urlImage}');
    margin-bottom: ${theme.spacings.medium};
  `}
`

export const InstructorNameAndProfessional = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      margin-bottom: calc(${theme.spacings.small} - 0.5rem);
      font-size: calc(${theme.font.sizes.medium} + 0.2rem);
    }

    @media ${theme.media.lteMedium} {
      text-align: center;
    }
  `}
`

export const InstructorProfessional = styled.span`
  ${({ theme }) => css`
    display: inline-block;
    color: ${theme.colors.gray7};
    font-size: ${theme.spacings.small};

    @media ${theme.media.lteMedium} {
      text-align: center;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`

export const AboutInstructor = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      flex-direction: column;
      justify-content: center;
    }
  `}
`
export const DescriptionInstructor = styled.p`
  ${({ theme }) => css`
    width: 100%;
    max-width: 624px;

    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.large};
  `}
`
export const LinkMoreInfo = styled.a`
  ${() => css`
    cursor: pointer;
  `}
`
