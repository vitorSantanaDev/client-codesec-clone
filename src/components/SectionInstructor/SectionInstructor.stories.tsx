import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionInstructor from '.'
import { mockDataSectionInstructor } from './mockData'

export default {
  title: 'SectionInstructor',
  component: SectionInstructor,
  args: {
    ...mockDataSectionInstructor
  }
} as ComponentMeta<typeof SectionInstructor>

export const Default: ComponentStory<typeof SectionInstructor> = (args) => (
  <SectionInstructor {...args} />
)
