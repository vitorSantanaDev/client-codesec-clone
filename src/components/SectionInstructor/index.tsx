import Link from 'next/link'

import Container from 'components/Container'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'

import { IViewProps } from './types'
import * as S from './styles'

const SectionInstructor = ({
  title,
  subtitle,
  instructor,
  description
}: IViewProps) => (
  <Container>
    <S.Wrapper>
      <S.AnimateWrapper data-aos="fade-right">
        <Subtitle>{subtitle}</Subtitle>
        <Heading>{title}</Heading>
      </S.AnimateWrapper>
      <S.InstructorInfoWrapper>
        <S.InstructorData>
          <S.InstructorPhoto
            data-aos="fade-down"
            urlImage={instructor.photo.urlImage || '/images/icon-autor.svg'}
          />
          <S.InstructorNameAndProfessional data-aos="fade-up">
            <Heading as="h5">{instructor.name}</Heading>
            <S.InstructorProfessional>
              Software Engineer
            </S.InstructorProfessional>
          </S.InstructorNameAndProfessional>
        </S.InstructorData>
        <S.AboutInstructor>
          <S.DescriptionInstructor data-aos="fade-left">
            {description}
          </S.DescriptionInstructor>
          <Link href={instructor.link.linkUrl}>
            <S.LinkMoreInfo>
              <Subtitle data-aos="fade-right">
                {instructor.link.linkText}
              </Subtitle>
            </S.LinkMoreInfo>
          </Link>
        </S.AboutInstructor>
      </S.InstructorInfoWrapper>
    </S.Wrapper>
  </Container>
)

export default SectionInstructor
