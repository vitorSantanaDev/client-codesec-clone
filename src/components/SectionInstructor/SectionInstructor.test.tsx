import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionInstructor from '.'
import { mockDataSectionInstructor } from './mockData'

describe('<SectionInstructor />', () => {
  it('should render title and subtitle', () => {
    renderTheme(<SectionInstructor {...mockDataSectionInstructor} />)
    const heading = screen.getByRole('heading', {
      name: /title example instructor Section/i
    })
    const subtitle = screen.getByText(/subtitle/i)

    expect(heading).toBeInTheDocument()
    expect(subtitle).toBeInTheDocument()
  })
  it('should render data of instructor', () => {
    renderTheme(<SectionInstructor {...mockDataSectionInstructor} />)
    const nameInstructor = screen.getByRole('heading', {
      name: /name of instructor/i
    })
    const description = screen.getByText(
      mockDataSectionInstructor.instructor.description
    )
    const urlPhotoInstructor = mockDataSectionInstructor.instructor.photoUrl

    expect(nameInstructor).toBeInTheDocument()
    expect(description).toBeInTheDocument()
    expect(urlPhotoInstructor).not.toBe(null)
  })
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <SectionInstructor {...mockDataSectionInstructor} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
