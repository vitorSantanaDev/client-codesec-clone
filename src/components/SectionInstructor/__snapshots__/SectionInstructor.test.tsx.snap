// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionInstructor /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c3 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c1 {
  width: 100%;
  padding: 6.4rem 0;
}

.c1 > .c2 {
  margin-bottom: calc(1.6rem - 0.8rem);
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 4.0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2.4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  width: 120px;
  height: 120px;
  border-radius: 50%;
  border: 2px solid;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url('/images/icon-autor.svg');
  margin-bottom: 2.4rem;
}

.c9 .c4 {
  margin-bottom: calc(1.6rem - 0.5rem);
  font-size: calc(2.4rem + 0.2rem);
}

.c10 {
  display: inline-block;
  color: #A3A3A3;
  font-size: 1.6rem;
}

.c12 {
  width: 100%;
  max-width: 624px;
  color: #A3A3A3;
  font-size: 1.6rem;
  margin-bottom: 3.2rem;
}

.c13 {
  cursor: pointer;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 > .c4 {
    font-size: calc(3.2rem + 0.4rem);
  }
}

@media (max-width:768px) {
  .c6 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c7 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c9 {
    text-align: center;
  }
}

@media (max-width:768px) {
  .c10 {
    text-align: center;
    margin-bottom: 2.4rem;
  }
}

@media (max-width:768px) {
  .c11 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class=""
      data-aos="fade-right"
    >
      <span
        class="c2 c3"
        color="redGradient"
      >
        Subtitle
      </span>
      <h1
        class="c4 c5"
        color="white"
      >
        Title Example Instructor Section
      </h1>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <div
          class="c8"
          data-aos="fade-down"
        />
        <div
          class="c9"
          data-aos="fade-up"
        >
          <h5
            class="c4 c5"
            color="white"
          >
            Name of Instructor
          </h5>
          <span
            class="c10"
          >
            Software Engineer
          </span>
        </div>
      </div>
      <div
        class="c11"
      >
        <p
          class="c12"
          data-aos="fade-left"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.
        </p>
        <a
          class="c13"
        >
          <span
            class="c2 c3"
            color="redGradient"
          >
            Saiba mais
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
